pkg/controllers/cluster/import.go

```diff
@@ -230,7 +260,19 @@ func (i *importHandler) importCluster(cluster *fleet.Cluster, status fleet.Clust
 		return status, err
 	}
 
-	if err := i.deleteOldAgent(cluster, kc, i.systemNamespace); err != nil {
+	if cluster.Spec.AgentNamespace != "" && cluster.Status.Agent.Namespace != agentNamespace {
+		// delete old agent if moving namespaces for agent
+		if err := i.deleteOldAgentBundle(cluster); err != nil {
+			return status, err
+		}
+		if cluster.Status.Agent.Namespace != "" {
+			if err := i.deleteOldAgent(cluster, kc, cluster.Status.Agent.Namespace); err != nil {
+				return status, err
+			}
+		}
+	}
+
+	if err := i.deleteOldAgent(cluster, kc, agentNamespace); err != nil {
 		return status, err
 	}
```

Comment:
Maybe we should consider deleting the bundle within the `deleteOldAgent` method for readability.

Reply:
I'm afraid of always deleting the bundle as this will only apply to the agent for the local cluster, not all clusters under management.

Reply:
I'm also thinking we could lock it behind a `bool`, but I'm good with the way it's currently written then.